<launch>
  <!-- Nodelet manager configuration -->
  <arg name="nodelet_manager_name" default="standalone_nodelet"/>
  <arg name="start_nodelet_manager" default="true"/>
  <arg name="stargazer_config" default="$(find stargazer_ros_tool)/res/stargazer.yaml"/>

  <!-- Nodelet manager (if enabled) -->
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager"
        if="$(arg start_nodelet_manager)"/>

  <node pkg="nodelet" type="nodelet" name="landmark_localizer_nodelet"
        args="load stargazer_ros_tool/LandmarkLocalizerInterfaceNodelet $(arg nodelet_manager_name)" output="screen">
    <param name="stargazer_config" value="$(arg stargazer_config)"/>
    <param name="pose_topic" value="/stargazer/camera_pose"/>
    <param name="landmark_topic" value="/landmarks_seen"/>
    <param name="map_frame" value="world"/>
    <param name="camera_frame" value="camera"/>
    <param name="robot_frame" value="vehicle"/>
    <param name="use_ceres" value="True"/>
    <param name="debug_mode" value="False"/>
  </node>

  <include file="$(find stargazer_ros_tool)/launch/static_transform.launch"/>
</launch>
